vault/data_source_azure_access_credentials.go

```diff
@@ -0,0 +1,243 @@
+			},
+			"client_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The client id for credentials to query the Azure APIs.",
+			},
+			"client_secret": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The client secret for credentials to query the Azure APIs.",
+			},
+			"lease_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Lease identifier assigned by vault.",
+			},
+			"lease_duration": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Lease duration in seconds relative to the time in lease_start_time.",
+			},
+			"lease_start_time": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Time at which the lease was read, using the clock of the system where Terraform was running",
+			},
+			"lease_renewable": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "True if the duration of this lease can be extended through renewal.",
+			},
+		},
+	}
+}
+
+func azureAccessCredentialsDataSourceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	backend := d.Get("backend").(string)
+	role := d.Get("role").(string)
+	path := backend + "/creds/" + role
+
+	data := make(map[string][]string)
+
+	log.Printf("[DEBUG] Reading %q from Vault with data %#v", path, data)
+	secret, err := client.Logical().ReadWithData(path, data)
+	if err != nil {
+		return fmt.Errorf("error reading from Vault: %s", err)
+	}
+	log.Printf("[DEBUG] Read %q from Vault", path)
+
+	if secret == nil {
+		return fmt.Errorf("no role found at path %q", path)
+	}
+
+	clientID := secret.Data["client_id"].(string)
+	clientSecret := secret.Data["client_secret"].(string)
+
+	d.SetId(secret.LeaseID)
+	_ = d.Set("client_id", secret.Data["client_id"])
+	_ = d.Set("client_secret", secret.Data["client_secret"])
+	_ = d.Set("lease_id", secret.LeaseID)
+	_ = d.Set("lease_duration", secret.LeaseDuration)
+	_ = d.Set("lease_start_time", time.Now().Format(time.RFC3339))
+	_ = d.Set("lease_renewable", secret.Renewable)
+
+	subscriptionID := d.Get("subscription_id").(string)
+	tenantID := d.Get("tenant_id").(string)
+
+	// If we're not supposed to validate creds, or we don't have enough
+	// information to do it, there's nothing further to do here.
+	validateCreds := d.Get("validate_creds").(bool)
+	if !validateCreds {
+		// We're done.
+		return nil
+	}
+	if subscriptionID == "" || tenantID == "" {
+		return fmt.Errorf(`if 'validate_creds' is true, both 'subscription_id' and 'tenant_id' must also be provided`)
+	}
+
+	// Let's, test the credentials before returning them.
+	vnetClient := network.NewVirtualNetworksClient(subscriptionID)
+
+	// You can't directly pass in a client ID and secret.
+	// The closest thing you can do is provide them through the environment.
+	revert, err := setAndPrepareReversion(auth.TenantID, tenantID)
```

Comment:
Maybe rename `setAndPrepareReversion` to something reflecting that it's updating env vars. "Prepare reversion" doesn't tell me much.

Reply:
Sure, I renamed it a bunch of times because I wasn't sure what to call it. I like the idea of saying something about env vars.

Reply:
https://github.com/terraform-providers/terraform-provider-vault/pull/713/commits/c2525e132a31c199cc614fc101220373b1b7e5e0